src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -186,7 +188,9 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
               bn(toBaseUnit(sendMax ? 0 : amountCryptoPrecision, asset.precision)).decimalPlaces(0),
             )
             .minus(estimatedFees.fast.txFee)
+            .minus(assetId === solAssetId ? solana.SOLANA_MINIMUM_RENT_EXEMPTION_LAMPORTS : 0)
```

Comment:
The meaty logic, keep at least the rent exemption amount in the balance so we don't close the account and sendMax is happy

Reply:
@NeOMakinG should this live in chain-adapters? We do have `sendMax` option which we leverage in `getFeeData()`, gut feel is this deduction should be part of returned fees?

Reply:
It looks like a good idea but it's actually not, it's not a fee as it's not deducted from the account, so I wouldn't add it to the fees

It might be able to live in `chainSpecific` though, but then it's not so interesting to add it to chain adapter

Reply:
@NeOMakinG rationale sounds sane to me, indeed it's not *payable* but contributes to the max balance you can send. Only thing is we have to
1. be cognicent that we'll have to add that in many different paths (at the very least, sends and trade execution if e.g sending max to Solana, think that's the only two out the top of my mind for the time being)
2. Comment out what this magic is about

@kaladinlight thoughts? Intuition tells me it feels like an antipattern not to do that in chain-adapters land and feels like overestimating for all other Txs would be a better solution (assuming this is only for max sends), but I think once you go past the intuition solution, the rationale with trading following patterns for correctness actually makes sense

Reply:
Swapper doesn't have any sendMax notion, and the rent exempt amount is very small, I think users won't hit it, and if they do, adding this to chain adapter if sendMax wouldn't prevent us to make it fail

This is kinda why it shouldn't leave in chain adapter as long as we can't remove this rent exempt from absolutely all txs regardless if it's sendmax, also chain adapter doesn't look like having the notion of the current balance, I feel like it's the responsibility of the consumer to handle this rent logic unfortunately, but happy to see the @kaladinlight input

Let's not block this PR, we can do a follow up anyway! As we would like to release Jupiter and bugfixes!